src/main/java/bot/CommandParser.java

```diff
@@ -0,0 +1,114 @@
+package bot;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * CommandParser class is concerned with parsing the input from the user.
+ * <p>
+ * Note that this class is not in fact responsible for the bot's individual commands (list command, todo command).
+ */
+public final class CommandParser {
+    /**
+     * The actual command the user inputted.
+     */
+    private final String command;
+
+    /**
+     * This description that came with this command.
+     */
+    private final String desc;
+
+    /**
+     * The extra arguments specified with the command.
+     */
+    private final Map<String, String> arguments;
+
+    /**
+     * Creates a Command object.
+     *
+     * @param command   The actual command inputted by the user.
+     * @param desc      Description for the command.
+     * @param artifacts Extra arguments specified with this command.
+     */
+    private CommandParser(String command, String desc, Map<String, String> artifacts) {
+        this.command = command;
+        this.desc = desc;
+        this.arguments = artifacts;
+    }
+
+    /**
+     * Creates a new Command from a user input line.
+     *
+     * @param input The inputted line from the console.
+     * @return New Command object with parsed information.
+     */
+    public static CommandParser fromString(String input) {
+        // Trim the input.
+        input = input.trim();
+
+        // We first split the command with a space. This will allow us to get the first word
+        // in the input. The first word will be the command that the user wants.
```

Comment:
You can consider keeping the comments at the top of the function instead of explaining throughout the code. Code should be self explanatory, don't have to explain how exactly the code works.